### PR TITLE
Split out speedtestdotnet into a sensor and component

### PR DESCRIPTION
**Pull request by rohankapoorcom**
**Description:**
Speedtest.net is now a component with a sensor platform.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#20527

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/sensor.speedtestdotnet.markdown [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+### {% linkable_title Using as a trigger in an automation %}
+
+{% raw %}
+```yaml
+# Example configuration.yaml entry
+automation:
+  - alias: "Internet Speed Glow Connect Great"
+    trigger:
+      - platform: template
+        value_template: "{{ states('sensor.speedtest_download')|float > 10 }}"
```
Comment (DubhAd) [inline source/_components/sensor.speedtestdotnet.markdown]: Neither of these match if the speed is exactly 10 ;)

**Review by DubhAd — DISMISSED** [review]

File: source/_components/sensor.speedtestdotnet.markdown (hunk 1 above)
Comment (rohankapoorcom) [reply source/_components/sensor.speedtestdotnet.markdown]: I didn't write this example, but I will fix it :)

**Comment by frenck** [comment]
Why split this out in multiple pages?
If you want it to be listed in multple categories, you could add them to the `ha_category` as a list:

```yaml
ha_category:
- System Monitor
- Sensor
```

We are trying to halt the growth of specific pages for each platform. Instead, we are trying to merge those pages into a single, more complete page. This PR does the opposite.

**Comment by rohankapoorcom** [comment]
This was done before I saw the new style, haven't had a chance to update.

**Comment by rohankapoorcom** [comment]
@frenck should be corrected now.

**Comment by frenck** [comment]
Thanks, @rohankapoorcom! 👍 

I've restored the original `ha_release` and `date`. Those indicate the initial creation of the document/feature.

**Review by frenck — APPROVED** [review]

**Comment by rohankapoorcom** [comment]
@frenck if a user tried the use the component as written in the docs before 0.88, it would not function because before 0.88, there was just a platform. Is it okay to have the original date and version then?

**Comment by frenck** [comment]
@rohankapoorcom Yep, we have breaking changes all the time.
Documentation always reflects the current release.

File: source/_components/speedtestdotnet.markdown [hunk 2]
```diff
@@ -91,39 +103,17 @@ test. You can turn on manual mode to disable the scheduled speed tests.
...
-# Example configuration.yaml entry
-sensor:
-  - platform: speedtest
-    minute: 30
-    hour:
-      - 0
-      - 6
-      - 12
-      - 18
+speedtest:
```
Comment (sqldiablo) [inline source/_components/speedtestdotnet.markdown:114]: Is this intentional? Isn't this called speedtestdotnet now?

File: source/_components/speedtestdotnet.markdown [hunk 3]
```diff
@@ -91,39 +103,17 @@ test. You can turn on manual mode to disable the scheduled speed tests.
...
+  update_interval:
+    minutes: 30
     monitored_conditions:
```
Comment (sqldiablo) [inline source/_components/speedtestdotnet.markdown:117]: This indenting is also wrong. monitored_conditions is a peer to update_interval, not a child.

File: source/_components/speedtestdotnet.markdown (hunk 2 above)
Comment (klaasnicolaas) [reply source/_components/speedtestdotnet.markdown:114]: Yes the component name has been changed

**Comment by rohankapoorcom** [comment]
Looks like @tjorim fixed both of those problems in (#8576). Thanks!

**Pull request merged**